pkg/execution/state/redis_state/queue.go

```diff
@@ -1979,6 +1987,17 @@ func (q *queue) Scavenge(ctx context.Context) (int, error) {
 			continue
 		}
 
+		p := QueuePartition{}
```

Comment:
Context: We can check whether we want to process the partition _before_ loading any items in the concurrency zset.